### PR TITLE
X509_verify_cert_error_string is exposed to go interface

### PR DESCRIPTION
**Pull request by MikeFalcon77**
GetVerifyError is implemented to get VerifyResult textual representation

### CODE REVIEW

**Comment by ssgreg** [comment]
Could you please define a separate error type with VerifyResult inside? It will help us in future to distinguish one of another.

**Comment by MikeFalcon77** [comment]
> Could you please define a separate error type with VerifyResult inside? It will help us in future to distinguish one of another.

Done

File: conn.go [hunk 1]
```diff
@@ -99,6 +99,25 @@ const (
 	ApplicationVerification       VerifyResult = C.X509_V_ERR_APPLICATION_VERIFICATION
 )
 
+type VerifyError struct {
```
Comment (ssgreg) [inline conn.go:102]: Use the code below:

```
type verifyError struct {
    code VerifyResult
}

func (e verifyError) Error() string {
    return fmt.Sprintf("openssl verification error: %s", C.GoString(C.X509_verify_cert_error_string(C.long(e.code))))
}

func NewVerifyError(code VerifyResult) error {
 	if code == Ok {
 		return nil
 	}

 	return &VerifyError{code: code}
 }
```

**Review by ssgreg — CHANGES_REQUESTED** [review]

File: conn.go (hunk 1 above)
Comment (MikeFalcon77) [reply conn.go:102]: Do you mean: 
type VerifyError struct {
    code VerifyResult
}
... 
?

But what's about Code() method? I would like to leave Code() for some cases

Also what is a problem to call GetVerifyError() in Error()?

Comment (ssgreg) [reply conn.go:102]: let's talk

**Review by ssgreg — APPROVED** [review]

**Pull request merged**